src/x509.cc

```diff
@@ -45,6 +45,85 @@ std::string parse_args(const Nan::FunctionCallbackInfo<v8::Value>& info) {
   return *String::Utf8Value(info[0]->ToString());
 }
 
+
+
+NAN_METHOD(verify) {
+  Nan::HandleScope scope;
+  OpenSSL_add_all_algorithms();
```

Comment:
That is true, but I suspect the code you pointed at only runs if the crypto module is required..

In fact if I remove the line the test fails with
 ```
"Error: certificate signature failure".
```
I am quite sure that the certificate signature is correct, because if I run openssl verify I get the expected result:

```
:~/code/node-x509$ openssl verify -CAfile test/CA_chains/enduser-example.com.chain test/certs/enduser-example.com.crt 
test/certs/enduser-example.com.crt: OK
```

If I add `require('crypto')` before the `verify()` function gets called,  everything is fine again.

(I obtained this result on node 4.4.3 and 6.2.2)

So I believe we have to choose to either require the crypto module in index.js or to keep that line. Maybe requiring crypto is safer? What do you suggest?